inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -283,6 +283,8 @@ public function save_post( $post_id ) {
 	 */
 	public function is_valid_post_type( $post_type ) {
 
+		global $wp_post_types;
```

Comment:
It is thin logically, but filter operation in it is far for free and it won't scale favorably with large amount of post types and large counts of this check firing. I am not talking ten or twenty here, this might fire _thousands_ of times in edge cases.

My opinion that if there is no functional or compat reason to use API here then accounting for performance takes priority.